packages/core/src/types.ts

```diff
@@ -36,25 +52,10 @@ export interface FileProblem {
 export interface SyntaxElement {
 	loc: SourceLocation;
 	range: [number, number];
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- This is a generic type
 	[key: string]: any;
```

Comment:
It's important to note that TS's [structural typing system](https://www.typescriptlang.org/docs/handbook/type-compatibility.html) means that having an index signature is not necessary if your goal is to _accept_ a subtype.

TS allows an assignment to a variable if the assignee has _at least_ the same properties as the receiving type.

The only time that an index signature like this is necessary is if you want to do "unsafe and untyped access to any property" - which is something you generally want to avoid because, as the description suggests, it's unsafe and untyped - defeating the purpose of having types at all!

Reply:
Yes, let's fix this separately.